src/utils/fs.ts

```diff
@@ -50,6 +50,34 @@ export function isFile (path: string): Promise<boolean> {
 }
 
 /**
+ * Verify a path exists and is a directory.
+ */
+export function isDirectory (path: string): Promise<boolean> {
+  return stat(path).then(stat => stat.isDirectory(), () => false)
+}
+
+function filterAsync<T>(items: T[], predicate: { (item: T): Promise<boolean> }): Promise<T[]> {
```

Comment:
Based on `array.prototype.filter()`, but supports predicates that return promises. Can move to a more general purpose util or get replaced by an external dep if more appropriate.